### PR TITLE
merge Cmake mkl branch

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,39 +92,46 @@ include_directories(${VOTCA_CSG_INCLUDE_DIRS})
...
+set(GSL TRUE) #used in votca_config.h
+endif()
+
+if(MKL_FOUND)
+  set(MKL_URL "http://software.intel.com/sites/default/files/managed/76/8c/intel-mkl-and-boost-example.zip")
+  set(MKL_MD5 B957EE63A2167AD21B650B47726CD55C )
+  set(MKL_FILE "intel-mkl-and-boost-example.zip")
+  
+  include(ExternalProject)
+  ExternalProject_add(intel
```
Comment (junghans) [inline CMakeLists.txt]: Is this actually needed or can we use the headers coming with VOTCA tools?

File: include/votca/xtp/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,9 @@
-configure_file(votca_xtp_config.h.in ${CMAKE_CURRENT_BINARY_DIR}/votca_xtp_config.h)
+configure_file(votca_config.h.in ${CMAKE_CURRENT_BINARY_DIR}/votca_config.h)
 
-file(GLOB_RECURSE VOTCA_HEADERS *.h ${CMAKE_CURRENT_BINARY_DIR}/votca_xtp_config.h)
+if (NOT GSL_FOUND)
+    file(GLOB_RECURSE NOT_VOTCA_HEADERS votca_gsl_boost_ublas_matrix_prod.h)
```
Comment (junghans) [inline include/votca/xtp/CMakeLists.txt]: Is this a duplication from the tools header?

File: include/votca/xtp/votca_config.h.in [hunk 3]
```diff
@@ -42,9 +42,9 @@
 
 
 #if defined(GSL)
-    #include <votca/xtp/gsl_boost_ublas_matrix_prod.h>
+    #include "gsl_boost_ublas_matrix_prod.h"
```
Comment (junghans) [inline include/votca/xtp/votca_config.h.in]: Can we not just use the matrix product header from tools?

File: CMakeLists.txt (hunk 1 above)
Comment (12AngryMen) [inline CMakeLists.txt]: okay,  bjoern agrees. We will try use tools as much as possible. So I ll remove that part and simply import the tools prod header

**Comment by junghans** [comment]
@12AngryMen : does this work with MKL?

**Comment by 12AngryMen** [comment]
@junghans it does work like a charm

**Pull request merged**